openml/flows/flow.py

```diff
@@ -1,80 +1,358 @@
+
+        Parameters
+        ----------
+        xml_dict : dict
+            Dictionary representation of the flow as created by _to_dict()
+
+        Returns
+        -------
+            OpenMLFlow
+
+        """
+        dic = xml_dict["oml:flow"]
+        flow_id = int(dic['oml:id']) if 'oml:id' in dic else None
+        uploader = dic.get('oml:uploader')
+        name = dic['oml:name']
+        external_version = dic.get('oml:external_version')
+        description = dic.get('oml:description')
+        upload_date = dic.get('oml:upload_date')
+        language = dic.get('oml:language')
+        dependencies = dic.get('oml:dependencies')
+        version = dic.get('oml:version')
+        binary_url = dic.get('oml:binary_url')
+        binary_format = dic.get('oml:binary_format')
+        binary_md5 = dic.get('oml:binary_md5')
+
+        parameters = OrderedDict()
+        parameters_meta_info = OrderedDict()
+        if 'oml:parameter' in dic:
+            if isinstance(dic['oml:parameter'], dict):
+                oml_parameters = [dic['oml:parameter']]
+            else:
+                oml_parameters = dic['oml:parameter']
+
+            for oml_parameter in oml_parameters:
+                parameter_name = oml_parameter['oml:name']
+                default_value = oml_parameter['oml:default_value']
+                parameters[parameter_name] = default_value
+
+                meta_info = OrderedDict()
+                meta_info['description'] = oml_parameter.get('oml:description')
+                meta_info['data_type'] = oml_parameter.get('oml:data_type')
+                parameters_meta_info[parameter_name] = meta_info
+
+        components = OrderedDict()
+        if 'oml:component' in dic:
+            if isinstance(dic['oml:component'], dict):
+                oml_components = [dic['oml:component']]
+            else:
+                oml_components = dic['oml:component']
+
+            for component in oml_components:
+                flow = OpenMLFlow._from_xml(component)
+                components[component['oml:identifier']] = flow
+
+        tags = []
+        if 'oml:tag' in dic and dic['oml:tag'] is not None:
+            if isinstance(dic['oml:tag'], dict):
+                oml_tags = [dic['oml:tag']]
+            else:
+                oml_tags = dic['oml:tag']
+
+            for tag in oml_tags:
+                tags.append(tag)
+
+        return cls(name=name, description=description, model=None,
+                   components=components, parameters=parameters,
+                   parameters_meta_info=parameters_meta_info,
+                   external_version=external_version,
+                   uploader=uploader, tags=tags, version=version,
+                   upload_date=upload_date, language=language,
+                   dependencies=dependencies, binary_url=binary_url,
+                   binary_format=binary_format, binary_md5=binary_md5,
+                   flow_id=flow_id)
+
+    def __eq__(self, other):
```

Comment:
This is so far only used in a test, and will later be used to check if the server stores flows correctly. I'm not sure if the user needs such an equality function. Should I make this private?